### DISCUSSION TITLE
How to bundle Emscripten output (ES6 module) with webpack?

### DISCUSSION THREAD

**Discussion by loryruta** [webpack/webpack#19005] [Q&A]
Hello,
I'm trying to understand how to bundle an app with webpack.
This app depends on a library (of mine) which in turns depends on .js+.wasm files coming from Emscripten compilation.

My project structure is the following:
```
./node_modules/library/
- module.mjs
- module.wasm
- index.mjs

./app/
- index.mjs --> depends on ../node_modules/library/index.mjs
- webpack.config.cjs
```

The module.mjs and module.wasm are Emscripten's output and are compiled with `MODULARIZE=1`, `EXPORT_ES6=1`.

The setup I've given to `library` is good for me as it enables the user to depend on `library` by using ES6 imports.

However, when the library is used on the web, for several reasons I would like to bundle the library using webpack.

I don't have preference over the output (one file vs multiple files), even having one single `bundle.js` file would be fine, just I'd like to see it working.

I'm facing several issues:
- emscripten puts node/web/worker code into one file. webpack tries to follow imports toward node packages (e.g. process) and fails. I've managed to avoid this using the `webpack.IgnorePlugin` to avoid following any import arose from `module.mjs`.\
Now the output is `bundle.js`, `js_lib_module-wasm_js.bundle.js`, `<hash>.wasm`.
- Now, almost everything is working, until `module.mjs` doesn't try to start the web worker. I've discovered webpack replaces `import.meta.url` with the path of the original file (e.g. `file://<parent>/node_modules/library/module.mjs`), and **not the webpack's output bundle**, and thus the call to `new Worker(new URL("file://<parent>/node_modules/library/module.mjs"))` fails.

I'm asking for any quick fix to this issue, but most importantly: isn't there a working example of app + ES6 Emscripten module? 
Thanks,

**Comment by alexander-akait** [comment]
@loryruta Can you create a reproducible repository? Want to see your configuration, thank you
